### PR TITLE
[subplugin/ext] Added support for setInputDim/reloadModel in unittests

### LINKED ISSUE #1959: Remove assert from extensions + add error checking testcases
Remove asserts on the return values of the functions used in tensor filter extensions.
This will allow adding unit-tests for failure scenarios.

### PR DESCRIPTION
**Pull request by kparichay**
Added check for setInputDim/reloadModel in unittests
Framework functions are reset at close in python subplugin
Framework functions are updated based on available in model file in custom subplugin

Resolves  #1959

Self evaluation:
Build test: [x]Passed [ ]Failed [ ]Skipped
Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor pk.kapoor@samsung.com

### CODE REVIEW

File: gst/nnstreamer/tensor_filter/tensor_filter_custom.c [hunk 1]
```diff
@@ -38,6 +38,7 @@ static void init_filter_custom (void) __attribute__ ((constructor));
 static void fini_filter_custom (void) __attribute__ ((destructor));
 
 static GstTensorFilterFramework NNS_support_custom;
```
Comment (dongju-chae) [inline gst/nnstreamer/tensor_filter/tensor_filter_custom.c]: let's remove this cuz its pointer is added.

Comment (kparichay) [reply gst/nnstreamer/tensor_filter/tensor_filter_custom.c]: Updated. Removed the added pointer, and kept this instead.

Comment (dongju-chae) [reply gst/nnstreamer/tensor_filter/tensor_filter_custom.c]: In fact, I think re-declaration of global variables is not good approach, and its compilation is prohibited in C++. But, anyway, this code works in C.

Comment (kparichay) [reply gst/nnstreamer/tensor_filter/tensor_filter_custom.c]: Yes, you are right, C++ gives compilation error. I removed the re-declaration, and used pointer style.

**Comment by myungjoo** [comment]
The commit  3bbe0f679321ec5979848f1aee9005dabb325ca2 appears to have a lot of changes not coherent with the commit message. Is it a rebase mistake?

**Comment by kparichay** [comment]
> The commit [3bbe0f6](https://github.com/nnsuite/nnstreamer/commit/3bbe0f679321ec5979848f1aee9005dabb325ca2) appears to have a lot of changes not coherent with the commit message. Is it a rebase mistake?

Rebased, but still there are a lot of changes - this is due reason that adding testcases of  `setInputDim` needed a lot of change. I will split this into two parts.

**Pull request merged**